python/graphstorm/model/utils.py

```diff
@@ -703,8 +703,85 @@ def shuffle_predict(predictions, id_mapping_file, pred_type,
                 len(predictions)).cpu() # predictions are stored in CPU
     return predictions[local_id_mapping]
 
+def shuffle_nids(g, ntype, nids, node_id_mapping_file, rank):
+    """ Shuffle node ids into the original node ids according to node_id_mappings
+
+        Parameters
+        ----------
+        g: Dist DGLGraph
+            Graph.
+        ntype: str
+            Node type of nids.
+        nids: torch.Tensor
+            Node ids.
+        node_id_mapping_file:
+            Path to the file storing node id mapping generated by the
+            graph partition algorithm.
+        rank: int
+            Rank of the current process in a distributed environment.
+    """
+    num_nodes = g.num_nodes(ntype)
+    id_mapping_info = create_dist_tensor((num_nodes,), dtype=th.int64,
+                                          name=f"mapping-{ntype}",
+                                          part_policy=g.get_node_partition_policy(ntype),
+                                          # TODO: this makes the tensor persistent in memory.
+                                          persistent=True)
```

Comment:
let's not turn on persistent here. are we sure that there is still a bug in DistDGL?

Reply:
Done.